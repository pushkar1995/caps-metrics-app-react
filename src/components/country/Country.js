import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { BsArrowRightCircle } from 'react-icons/bs';
import {
  getAllCountries,
  getRegion,
} from '../../redux/countriesSlice/countriesSlice';
import './Country.scss';

const Country = () => {
  const { countriesData, isLoading, region } = useSelector(
    (store) => store.country,
  );
  const dispatch = useDispatch();

  useEffect(() => {
    if (!countriesData.length) dispatch(getAllCountries());
    if (region) dispatch(getRegion(region));
  }, [dispatch, countriesData.length, region]);

  return (
    <div className="country__list">
      {isLoading ? (
        <h1>Loading...</h1>
      ) : (
        countriesData.length > 0
        && countriesData.map((item) => (
          <Link
            to={`/${item.cioc}`}
            key={item.name.common}
            className="country__container"
          >
            <img src={item.flags.png} alt={item.flags.alt} />
            <div className="country__content">
              <BsArrowRightCircle className="country__arrow" />
              <h3>
                Name:
                {item.name.common}
              </h3>
              <p>
                Capital:
                {item.capital}
              </p>
              <p>
                Population:
                {item.population}
              </p>
            </div>
          </Link>
        ))
      )}
    </div>
  );
};

export default Country;
