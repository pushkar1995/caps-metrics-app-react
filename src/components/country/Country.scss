.country__list {
  display: grid;
  color: var(--white-color);
  grid-template-columns: 1fr 1fr;

  .country__container {
    position: relative;
    padding: 0.6rem;
    background-color: var(--dark-secondary-color);

    img {
      height: 100px;
      width: 150px;
      border-radius: 8px;
    }

    .country__content {
      display: flex;
      justify-content: center;
      flex-direction: column;
      font-family: 'Roboto Slab', serif;
      gap: 0.5rem;
      color: #ebebeb;
    }
  }

  .country__arrow {
    font-size: 22px;
    display: flex;
    position: absolute;
    top: 5px;
    right: 5px;
  }

  .country__container:nth-child(3n+1) {
    background-color: var(--secondary-color);
  }
}
