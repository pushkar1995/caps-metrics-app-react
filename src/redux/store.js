import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countriesSlice/countriesSlice';

const store = configureStore({
  reducer: {
    country: countriesReducer,
  },
});

export default store;
