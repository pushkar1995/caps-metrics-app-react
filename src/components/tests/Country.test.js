import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Country from '../country/Country';
import store from '../../redux/store';

describe('Test the Country Component', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Country />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
