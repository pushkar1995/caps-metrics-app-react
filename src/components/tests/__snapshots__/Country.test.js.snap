// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Country Component should match to snapshot 1`] = `
<div
  className="country__list"
/>
`;
